src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs

```diff
@@ -1489,7 +1519,7 @@ internal static bool CanStoreTypeOnStack(TypeDesc type)
         {
             if (type is DefType defType)
             {
-                if (!defType.IsGCPointer && !defType.ContainsGCPointers)
+                if (!defType.IsGCPointer && !defType.ContainsGCPointers && !ContainsIsByReferenceOfT(type))
```

Comment:
`CORINFO_FLG_CONTAINS_STACK_PTR` is equivalent to `type.IsByRefLike`. Should this just call `type.IsByRefLike`?

Reply:
It is correct with `type.IsByRefLike`  so I guess so, I just couldn't see where it did the recursion, i.e if a field is `IsByRefLike`, then so is it's struct.

Reply:
> `CORINFO_FLG_CONTAINS_STACK_PTR` is equivalent to `type.IsByRefLike`

For my education: this means that reflection's `IsByRefLike` is not equivalent to type system's `IsByRefLike`, correct?

Reply:
Reflection IsByRefLike is same as type system IsByRefLike. And the type system IsByRefLike is same as `CORINFO_FLG_CONTAINS_STACK_PTR`.

Using `IsByRefLike` / `CORINFO_FLG_CONTAINS_STACK_PTR` here is a simplification that is leaving some performance on the table for byref-like types that do not need GC tracking.

Reply:
> CORINFO_FLG_CONTAINS_STACK_PTR is equivalent to type.IsByRefLike. Should this just call type.IsByRefLike?

I'm finding these aren't quite the same, and `CORINFO_FLG_CONTAINS_STACK_PTR` is more like what I want.

```cs
    public ref struct GenStructWithImplicitOp<TKey>
    {
        private int length;
        private int length2; // just one int field will not create an LLVM struct type, so put another field

        public GenStructWithImplicitOp(TKey[] key)
        {
            length = key.Length;
            length2 = length;
        }

        public static implicit operator GenStructWithImplicitOp<TKey>(TKey[] array) => new GenStructWithImplicitOp<TKey>(array);
    }
```
With `GenStructWithImplicitOp<string>`  .   This struct is `IsByRefLike` but `CORINFO_FLG_CONTAINS_STACK_PTR` is not set.  Is `IsByRefLike` being set because its a  `ref struct`?

Reply:
Yes, both `IsByRefLike` and `CORINFO_FLG_CONTAINS_STACK_PTR` means `ref struct`. It means that these can be set even for structs that do not require any GC tracking, like the struct from your example.

If you need precise answer for whether the struct requires GC tracking, algorithm like what you have in `ContainsIsByReferenceOfT` or the `HasGCPtr` property in the JIT is way to go.